src/test/scala/mesosphere/marathon/api/akkahttp/v2/AppsControllerTest.scala

```diff
@@ -0,0 +1,1856 @@
+          |    }]
+        """.stripMargin, groupManager, auth
+      )
+
+      Post(Uri./, entity) ~> route ~> check {
+        Then("The return code indicates create success")
+        status shouldEqual StatusCodes.Created
+        header[Headers.`Marathon-Deployment-Id`] should not be 'empty
+      }
+    }
+
+    "Creating an app with an external volume and MESOS containerizer should pass validation" in new Fixture {
+      Given("An app with a named, non-'agent' volume provider")
+      val entity = createAppWithVolumes(
+        "MESOS",
+        """
+          |    "volumes": [{
+          |      "containerPath": "var",
+          |      "external": {
+          |        "size": 10,
+          |        "provider": "dvdi",
+          |        "name": "namedfoo",
+          |        "options": {"dvdi/driver": "bar"}
+          |      },
+          |      "mode": "RW"
+          |    }]
+        """.stripMargin, groupManager, auth
+      )
+
+      Then("The return code indicates create success")
+      Post(Uri./, entity) ~> route ~> check {
+        Then("The return code indicates create success")
+        status shouldEqual StatusCodes.Created withClue s"response=${responseAs[String]}"
+        header[Headers.`Marathon-Deployment-Id`] should not be 'empty
+      }
+    }
+
+    "Creating an app with an external volume using an invalid rexray option should fail" in new Fixture {
+      Given("An app with a named, non-'agent' volume provider")
+      val entity = createAppWithVolumes(
+        "MESOS",
+        """
+          |    "volumes": [{
+          |      "containerPath": "var",
+          |      "external": {
+          |        "size": 10,
+          |        "provider": "dvdi",
+          |        "name": "namedfoo",
+          |        "options": {"dvdi/driver": "rexray", "dvdi/iops": "0"}
+          |      },
+          |      "mode": "RW"
+          |    }]
+        """.stripMargin, groupManager, auth
+      )
+
+      Then("The return code indicates validation error")
+      Post(Uri./, entity) ~> route ~> check {
+
+        status shouldEqual StatusCodes.UnprocessableEntity
+        responseAs[String] should include("/container/volumes(0)/external/options/\\\"dvdi/iops\\\"")
+      }
+    }
+
+    "Creating an app with an external volume w/ relative containerPath DOCKER containerizer should succeed (available with Mesos 1.0)" in new Fixture {
+      Given("An app with a named, non-'agent' volume provider")
+      val entity = createAppWithVolumes(
+        "DOCKER",
+        """
+          |    "volumes": [{
+          |      "containerPath": "relative/path",
+          |      "external": {
+          |        "provider": "dvdi",
+          |        "name": "namedfoo",
+          |        "options": {"dvdi/driver": "bar"}
+          |      },
+          |      "mode": "RW"
+          |    }]
+        """.stripMargin, groupManager, auth
+      )
+
+      Then("The return code indicates create success")

```

Comment:
I think we should check the rejections instead. Let's decide tomorrow to follow up. I don't see this as a blocker for this diff.